doc/lua_api.txt

```diff
@@ -4246,6 +4246,8 @@ Call these functions only at load time!
       the puncher to the punched.
     * `damage`: Number that represents the damage calculated by the engine
     * should return `true` to prevent the default damage mechanism
+* `minetest.register_on_rightclickplayer(function(player, clicker))`
```

Comment:
It might be worth providing more data to this callback like clicker's wielded item

Reply:
This might lead to less intuitive mods being built on the top of it, if we assume that "left-click" means "apply wielded tool to". Consider that the "on_rightclick" luaentity callback does not pass anything about the wielded item, while "on_punch" gets tool capabilities passed.

Reply:
> This might lead to less intuitive mods being built on the top of it, if we assume that "left-click" means "apply wielded tool to".

Sorry, I'm unable to understand that correctly - what does passing wielded item have anything to do with less intuitive mods?

Reply:
Mod developers might be more tempted to write a right-click handler for actions that involve the wielded item if it gets passed, even though it would be more intuitive for such actions to be triggered by a left-click.

In the end, mod developers will always be able to build what they want, if necessary by determining the wielded item through a separate API call. But if it was intended, I guess "on_rightclick" would also pass more data.

Reply:
Yeah, if mods want a right-click handler involving the wielded item, they can just call `clicker:get_wielded_item`. I also don't see what wrong with mods wanting to implement right-click handlers if they want to. :)

All I'm suggesting is for this to be provided by the callback itself, because

- convenience - shaves of *at least* one line from the callbacks that require retrieving the clicker's wielded item.
- performance - Wielded item would be retrieved in C++, which is going to be infinitesimally faster than manually calling `clicker:get_wielded_item()`. This might add up when there are a lot of player-rightclicks though.
- consistency with the rest of the API - most, if not all, interaction-based callbacks provide the wielded item along with other callback-specific data.

Reply:
That's an interesting suggestion. However, rightclicking nodes does not provide the wielded item either :)

I feel that it would be more sensible to keep this callback as general as possible. There's no way to anticipate what information will be needed for every use-case. I may only want to perform an action when the other player is directly in front of me and facing toward me. That means 'look_dir' and 'pos' should be passed as well. Where then do we draw the line?

Reply:
That's exactly why I didn't suggest adding `look_dir` and `pos`. I suggested wielded item as it's pretty generic and used by most callbacks. :)